gcn_prot/models/layers.py

```diff
@@ -41,3 +41,31 @@ def __repr__(self):
         return (
             f"{self.__class__.__name__} " f"({self.in_features} -> {self.out_features})"
         )
+
+class NormallizationLayer(nn.Module):
+    """Simple GCN layer.
+
+    Adaped from https://github.com/CrivelliLab/Protein-Structure-DL/
+    """
+
+    def __init__(
+        self, in_features, out_features, dropout=0.1, bias=False, act=F.relu, D=100.0
+    ):
+        """Initialize layer."""
+        super(NormallizationLayer, self).__init__()
+        self.weight_feat = out_features
+        # Define trainable parameters
+        self.weight1 = nn.Linear(in_features, self.weight_feat, bias)
+        self.weight2 = nn.Linear(in_features, self.weight_feat, bias)
+        self.d = D
+
+    def forward(self, input):
+        """Pass forward features `v` and sparse adjacency matrix `adj`."""
+        v, adj = input
+        v_shape = v.shape
+        if len(v_shape) > 2:
```

Comment:
This was done in the `GraphConvolution` to flatten the feature matrix `v` over the batch axis when there was effectively a batch dimension. I don't think that this will be needed here.

Reply:
It is now reflected in the tests that the online learning (no batch dimension) should also work.